app/questionnaire/path_finder.py

```diff
@@ -44,7 +44,21 @@ def routing_path(
         if section:
             for group in section["groups"]:
                 if "skip_conditions" in group:
-                    if evaluate_skip_conditions(
+                    if isinstance(group.get("skip_conditions"), dict):
+                        when_rule_evaluator = WhenRuleEvaluator(
+                            self.schema,
+                            self.answer_store,
+                            self.list_store,
+                            self.metadata,
+                            self.response_metadata,
+                            location=current_location,
+                            routing_path_block_ids=routing_path_block_ids,
+                        )
```

Comment:
We should create this outside the loop so that it isn't initialised multiple times. It would be temporarily we would create with regardless of if it is new/old, but I think that is fine as the overhead is negligible and makes it consistent with routing rules.

Reply:
Moved it above the `for` loop, I hope that is what you meant, thanks!